lib/ranked-model.rb

```diff
@@ -11,6 +11,7 @@ class NonNilColumnDefault < StandardError; end
   MIN_RANK_VALUE = -2147483648
 
   def self.included base
+    super
```

Comment:
this is really a good idea to have. oh well

Reply:
I'm unfamiliar with this technique. Can you link me to something that explains why it's necessary or nice to have?

Reply:
I mean, it may be unexpected but it makes sense, too. It's just Ruby

```sh
ruby <<EOF
class Foo
  attr_reader :bar
  def self.inherited(base)
    puts "I am in Foo: #{base}"
  end
end
class Bar < Foo
  def self.inherited(base)
    puts "I am in Bar: #{base}"
  end
end
class Baz < Bar
  def self.inherited(base)
    puts "I am in Baz: #{base}"
  end
end
EOF
```

prints

```
I am in Foo: Bar
I am in Bar: Baz
```

while

```sh
ruby <<EOF
class Foo
  attr_reader :bar
  def self.inherited(base)
    super
    puts "I am in Foo: #{base}"
  end
end
class Bar < Foo
  def self.inherited(base)
    super
    puts "I am in Bar: #{base}"
  end
end
class Baz < Bar
  def self.inherited(base)
    super
    puts "I am in Baz: #{base}"
  end
end
EOF
```

prints


```
I am in Foo: Bar
I am in Foo: Baz
I am in Bar: Baz
```

and

```sh
ruby <<EOF
class Foo
  attr_reader :bar
  def self.inherited(base)
    puts "I am in Foo: #{base}"
    super
  end
end
class Bar < Foo
  def self.inherited(base)
    puts "I am in Bar: #{base}"
    super
  end
end
class Baz < Bar
  def self.inherited(base)
    puts "I am in Baz: #{base}"
    super
  end
end
EOF
```

prints

```
I am in Foo: Bar
I am in Bar: Baz
I am in Foo: Baz
```

Reply:
Your examples are around `self.inherited` but this method is `self.included`. Am I missing something? :)

Reply:
So to clarify, this is just in case someone chooses to extend this module for some reason? :)

Reply:
well, no, because you `include RankedModel` in an ActiveRecord class, so it can affects what else is included.  I tried to find the AMS bug report where this came up but haven't found it yet, which is why I wrote that example. I'll write a new one

Reply:
Hehe, that's what I was thinking. Not sure why anyone would want to extend RankedModel. And is it possible to extend a module?

Reply:
you can include or prepend a module into a module :)  the only way to inherit a module is by fooling Ruby because Module is a class

```ruby
class SomeModule < Module
end
include SomeModule
```

Reply:
Lol, I'm glad we've managed to nut this one out :) So I guess remove this line from the PR?

Reply:
I guess there's https://github.com/rubocop-hq/ruby-style-guide/issues/809 https://github.com/rubocop-hq/rubocop/pull/8376 https://github.com/rubocop-hq/rubocop/commit/9d5cbe6e51f498f49152132518944ecb952d6f67

Reply:
Yep sounds like they don't raise anything on module `included` based on my quick skim of that code. I assume that's what you were saying?

Are you happy to remove this and also look at calling `connection` directly on the class per our discussions below and see how it tests out?